### ISSUE TITLE
GradCam implementation 

### ISSUE THREAD

**Issue by LukeWood** [keras-team/keras-cv#171]
https://keras.io/examples/vision/grad_cam/

The call signature should be something like:

overlaid_result_images = gradcam(model, images, classes)

**Comment by innat** [comment]
Relevant issue: https://github.com/keras-team/keras-cv/issues/53

Under which directory the visualization tools will be live? 

@MrinalTyagi you may like to contribute this. https://github.com/keras-team/keras/issues/15704

**Comment by MrinalTyagi** [comment]
I feel like creating a new directory named visualization would be a good option as it would be a separate section. If that's not the case adding it to utils can also be a good option.

**Comment by innat** [comment]
or, maybe a directory like `keras_cv.utils.grad_cam`.

**Comment by MrinalTyagi** [comment]
@innat @LukeWood Is it possible to delay this feature for some time as I wanted to propose this as an idea submission for this year's GSoC. Here is a basic draft as well for the proposal.  https://docs.google.com/document/d/1k6vI4X1INUUsbVwL5E2fgKy8UgT0mRKKFlDY-LB7iSE/edit?usp=sharing . Wanted to propose the addition of a whole visualization section in the project. Do let me know if you guys are fine with it.

**Comment by LukeWood** [comment]
I think visualization is deserving of its own directory

**Comment by MrinalTyagi** [comment]
@LukeWood Would you allow me to work on this during GSoC? I have updated the above proposal deck to be more in TensorFlow template format. Do have a look at it and let me know if I can take this whole visualization section during GSoC. Apart from that if you agree or not I would love to work on the same. Thanks for your time.

**Comment by bhack** [comment]
@lucasdavid I think you could start somewhere with a PR. 
as @MrinalTyagi  has still https://github.com/keras-team/keras-cv/pull/175 under early review.

**Comment by innat** [comment]
@lucasdavid Here is one reference, https://github.com/keisen/tf-keras-vis. It may help.

**Comment by LukeWood** [comment]
> @LukeWood Would you allow me to work on this during GSoC? I have updated the above proposal deck to be more in TensorFlow template format. Do have a look at it and let me know if I can take this whole visualization section during GSoC. Apart from that if you agree or not I would love to work on the same. Thanks for your time.

Thanks for the proposal, we really appreciate it @MrinalTyagi.  There are many visualization tools available that we can write and implement, so even if one is picked up (GradCam) by a contributor we can still find others to keep adding.  there are many with unique value.  Thanks for your understanding.

**Comment by LukeWood** [comment]
> I would also very much like to contribute in this part, if you need anything @MrinalTyagi.
> 
> If it is of any help, I have implemented Gradients, Smooth-Grad, Full-Grad, Guided-Grad, Grad-CAM, Grad-CAM++, and Score-CAM in [this repo](https://github.com/lucasdavid/algorithms-in-tensorflow/tree/main/notebooks/explaining), although they need a little polish to work with multiple explaining units at once (namely, to use `tf.batch_jacobian` instead of `tf.gradient`).

@lucasdavid can you open up unique github issues for each of these components?  I'd love to begin collecting visualization components that we would like to add.  Thanks a lot for your help
